src/qspi.rs

```diff
@@ -0,0 +1,324 @@
+                        for i in 0..num_unpack {
+                            buf[idx] = ((word & (0xFF << i * 8)) >> i * 8).try_into().unwrap();
+                            idx += 1;
+                        }
+                    }
+                }
+            }
+            None => (),
+        };
+
+        Ok(())
+    }
+
+    /// Polling indirect write.
+    pub fn polling_write(&mut self, buf: &[u8], transaction: QspiTransaction) -> Result<(), Error> {
+        // Clear DMA bit since we are not using it
+        self.qspi.cr.modify(|_, w| w.dmaen().clear_bit());
+
+        // Setup the transaction registers
+        self.setup_transaction(QspiMode::INDIRECT_WRITE, &transaction);
+
+        // If the transaction has data, write it word-by-word to the data register
+        match transaction.data_len {
+            Some(len) => {
+                let mut idx: usize = 0;
+                while idx < len {
+                    // Check if the FIFO is empty
+                    let num_bytes = self.qspi.sr.read().flevel().bits();
+                    if num_bytes == 0 {
+                        // Pack the word
+                        let mut word: u32 = 0;
+                        let num_pack = if (len - idx) >= 4 { 4 } else { len - idx };
+                        for i in 0..num_pack {
+                            word |= (buf[idx] as u32) << (i * 8);
+                            idx += 1;
+                        }
+
+                        // Write a word
+                        unsafe {
+                            self.qspi.dr.write(|w| w.data().bits(word));
+                        }
+                    }
+                }
+            }
+            None => (),
+        };
+
+        Ok(())
+    }
+
+    /// Map from QspiTransaction to QSPI registers.
+    fn setup_transaction(&mut self, fmode: u8, transaction: &QspiTransaction) {
+        unsafe {
+            // Clear any prior status flags
+            self.qspi.fcr.write(|w| w.bits(0x1B));
+
+            // Update data length, if applicable
+            match transaction.data_len {
+                Some(len) => self.qspi.dlr.write(|w| w.bits(len as u32 - 1)),
+                None => (),
+            };
+
+            // Update CCR register with metadata
+            self.qspi.ccr.write_with_zero(|w| {
+                w.fmode()
+                    .bits(fmode)
+                    .imode()
+                    .bits(transaction.iwidth)
+                    .admode()
+                    .bits(transaction.awidth)
+                    .dmode()
+                    .bits(transaction.dwidth)
+                    .adsize()
+                    .bits(self.adsize)
+                    .abmode()
+                    .bits(QspiWidth::NONE)
+                    .dcyc()
+                    .bits(transaction.dummy)
+                    .instruction()
+                    .bits(transaction.instruction)
```

Comment:
It would be nicer to group each field access into a separate statement, like this:
``` rust
w.fmode().bits(fmode);
w.imode().bits(transaction.iwdidth);
// ...
```

That's much more readable, in my opinion, as it prevents rustfmt from messing up the grouping between selecting the field and writing to the field.

Reply:
Yes I agree that is much more readable. I was trying to group the field and bits statements onto 1 line but rustfmt kept moving them - I think the issue was that I was not repeating `w`.